### PR TITLE
Bump googletest for compatibility with clang-cl

### PR DESCRIPTION
**Pull request by sunjayBhatia**
Commit Message: Bump googletest

For compatibility with Windows clang-cl
Additional Description: Submitting this independently to determine that this revision of googletest does not introduce breakage on any toolchain. See issue: https://github.com/google/googletest/issues/2490. Step to address https://github.com/envoyproxy/envoy/issues/11974
Risk Level: Low
Testing: Compiles with Windows clang-cl
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

**Comment by sunjayBhatia** [comment]
cc @wrowe

**Comment by wrowe** [comment]
@htuch This is the googletest bump required for clang-cl on Windows. Let's see how it pans out on all existing CI pipelines, then please expedite merging to master on success.

File: bazel/repository_locations.bzl
Comment (htuch) [inline bazel/repository_locations.bzl:436]: Thanks for the update. Please include an explanation as per policy, see https://github.com/envoyproxy/envoy/blob/master/DEPENDENCY_POLICY.md

> Versions should prefer release versions over master branch GitHub SHA tarballs. A comment is necessary if the latter is used. This comment should contain the reason that a non-release version is being used and the YYYY-MM-DD when the last update was performed.

**Comment by wrowe** [comment]
As this is a blocking issue for other PR's we would ask @envoyproxy/senior-maintainers if anyone has cycles to give this a green light and pull it?

File: bazel/repository_locations.bzl
Comment (mattklein123) [reply bazel/repository_locations.bzl:436]: @wrowe ping can you add the comment?

/wait

Comment (wrowe) [reply bazel/repository_locations.bzl:438]: Indeed, I had misunderstood the exception r.e. CPE's and test-only elements of the build, but hope you find the new message appropriate. Not really certain if there will be a 1.11 "release" given the general google philosophy of consuming the dev branch, and with respect to our test architecture, any of their bleed efforts to identify defects should be picked up fairly frequently (although probably not daily.)

**Review by mattklein123 — APPROVED** [review]
Thanks!

**Review by stonigrl — COMMENTED** [review]
🤞

**Pull request merged**
